src/other/index.jl

```diff
@@ -221,6 +221,17 @@ end
     isempty(idx.cols) ? (1:length(x)) : throw(ArgumentError("All(args...) is not supported: use Cols(args...) instead"))
 @inline Base.getindex(x::AbstractIndex, idx::Cols) =
     isempty(idx.cols) ? Int[] : union(getindex.(Ref(x), idx.cols)...)
+@inline Base.getindex(x::AbstractIndex, idx::Cols{Tuple{typeof(:)}}) = x[:]
+@inline Base.getindex(x::AbstractIndex, idx::Cols{<:Tuple{Function}}) =
+    findall(idx.cols[1], names(x))
+
+"""
+    Cols(f::Function)
+
+Select the columns whose names passed to the `f` predicate as strings return `true`.
+As a special case if `:` is passed (the `Colon` function) select all columns.
+"""
+Cols
```

Comment:
Shouldn't this be documented in the `select`, etc. docstrings?

Reply:
I think we should add this docstring. Apart from this all special selectors now are only mentioned in https://github.com/JuliaData/DataFrames.jl/blob/main/docs/src/lib/indexing.md. We should decide on a good place to document them in general, especially when we will start allowing broadcasting them. Do you see any good place?

Reply:
@nalimilan I tried to improve the documentation. The PR should be good to re-review now.

Reply:
We really need page(s) about indexing and `select`/`transform`/... which would be more detailed than Getting Started, but less technical that the API reference.

It's a bit weird that we add a docstring to `Cols` that only makes sense for DataFrames. Maybe we should document this in DataAPI?

Reply:
I have opened https://github.com/JuliaData/DataAPI.jl/pull/42 (and then we can remove the docstring here)
and https://github.com/JuliaData/DataFrames.jl/issues/2887 (to handle this in the future)